### PR TITLE
chore: support historical data points

### PR DESCRIPTION
**Pull request by TanZng**
note: some tests are failling

### CODE REVIEW

**Review by guelfey — COMMENTED** [review]
can you adjust the buildkite setup such that the tests make the pipeline fails and take a look why the tests are failing? Also for reviewing it would be helpful to split the changes into separate commits: what was from the original upstream PR and what changes did you do afterwards?

File: pkg/job/discovery_test.go [hunk 1]
```diff
@@ -425,9 +421,6 @@ func Test_getFilteredMetricDatas(t *testing.T) {
 				if *got.AddCloudwatchTimestamp != *tt.wantGetMetricsData[i].AddCloudwatchTimestamp {
 					t.Errorf("getFilteredMetricDatas().AddCloudwatchTimestamp = %v, want %v", *got.AddCloudwatchTimestamp, *tt.wantGetMetricsData[i].AddCloudwatchTimestamp)
 				}
-				if *got.AddHistoricalMetrics != *tt.wantGetMetricsData[i].AddHistoricalMetrics {
```
Comment (guelfey) [inline pkg/job/discovery_test.go]: why do we remove this test check again?

Comment (TanZng) [reply pkg/job/discovery_test.go]: `AddHistoricalMetrics` is not in the model anymore but is pass as a var in mapResultsToMetricDatas

Comment (guelfey) [reply pkg/job/discovery_test.go]: but it's still present in the model here? https://github.com/grid-x/yet-another-cloudwatch-exporter/blob/d7147db6a6ad9349fb75867b29041b00dba20765/pkg/model/model.go#L56
I'm not too deep into the code - it just seems weird to me to add this field explicitly to all the expected test data, but then not actually check whether the field matches the expected value. If it doesn't matter, then we should remove initializing it e.g. here https://github.com/grid-x/yet-another-cloudwatch-exporter/blob/d7147db6a6ad9349fb75867b29041b00dba20765/pkg/job/discovery_test.go#L93; if it does matter and we expect it to be `false`, we should re-add this line here

Comment (TanZng) [reply pkg/job/discovery_test.go]: true, in the PR `AddHistoricalMetrics` was a `bool` not a `*bool`. So it was false by default, now the value passed throw the functions @guelfey

**Review by guelfey — APPROVED** [review]

**Pull request merged**